src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -96,7 +97,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     Entity lectureEntity = LectureUtil.createEntity(lectureName, videoUrl, videoId.get());
     datastore.put(lectureEntity);
     try {
-      initializeTranscript(lectureEntity);
+      String transcriptLanguage =
+          Optional.ofNullable(request.getParameter(PARAM_LANGUAGE)).orElse("");
+      initializeTranscript(lectureEntity, transcriptLanguage);
```

Comment:
Why not pass in Optional<String> for transcript language to initializeTranscript? In that method you can set the empty string. This encapsulates the logic of what to set when there is no language.

Reply:
Gotcha, I originally figured it would be nice to extract the value as early as possible, but encapsulating it makes sense. Updated!